tests/bsim/bluetooth/audio/src/public_broadcast_source_test.c

```diff
@@ -0,0 +1,381 @@
+		return err;
+	}
+
+	/* Setup periodic advertising data */
+	err = bt_cap_initiator_broadcast_get_base(source, &base_buf);
+	if (err != 0) {
+		printk("Failed to get encoded BASE: %d\n", err);
+		return err;
+	}
+
+	per_ad.type = BT_DATA_SVC_DATA16;
+	per_ad.data_len = base_buf.len;
+	per_ad.data = base_buf.data;
+	err = bt_le_per_adv_set_data(adv, &per_ad, 1);
+	if (err != 0) {
+		printk("Failed to set periodic advertising data: %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int start_extended_adv(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	/* Start extended advertising */
+	err = bt_le_ext_adv_start(adv, BT_LE_EXT_ADV_START_DEFAULT);
+	if (err) {
+		printk("Failed to start extended advertising: %d\n", err);
+		return err;
+	}
+
+	/* Enable Periodic Advertising */
+	err = bt_le_per_adv_start(adv);
+	if (err) {
+		printk("Failed to enable periodic advertising: %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int setup_extended_adv(struct bt_le_ext_adv **adv)
+{
+	int err;
+
+	/* Create a non-connectable non-scannable advertising set */
+	err = bt_le_ext_adv_create(BT_LE_EXT_ADV_NCONN_NAME, NULL, adv);
+	if (err != 0) {
+		printk("Unable to create extended advertising set: %d\n", err);
+		return err;
+	}
+
+	/* Set periodic advertising parameters */
+	err = bt_le_per_adv_set_param(*adv, BT_LE_PER_ADV_DEFAULT);
+	if (err) {
+		printk("Failed to set periodic advertising parameters: %d\n",
+		       err);
+		return err;
+	}
+
+	return 0;
+}
+
+static int stop_extended_adv(struct bt_le_ext_adv *adv)
+{
+	int err;
+
+	err = bt_le_per_adv_stop(adv);
+	if (err) {
+		printk("Failed to stop periodic advertising: %d\n", err);
+		return err;
+	}
+
+	err = bt_le_ext_adv_stop(adv);
+	if (err) {
+		printk("Failed to stop extended advertising: %d\n", err);
+		return err;
+	}
+
+	err = bt_le_ext_adv_delete(adv);
+	if (err) {
+		printk("Failed to delete extended advertising: %d\n", err);
+		return err;
+	}
+
+	return 0;
+}
+
+static struct bt_bap_stream_ops broadcast_stream_ops = {
+	.started = started_cb,
+	.stopped = stopped_cb,
+	.sent = sent_cb
+};
+
+static void test_main(void)
+{
+	int err;
+	int count = 0;
+	uint8_t pba_params = 0;
+
+	err = bt_enable(NULL);
+	if (err) {
+		FAIL("Bluetooth enable failed (err %d)\n", err);
+		return;
+	}
+
+	broadcast_stream = &broadcast_source_stream;
+	bt_bap_stream_cb_register(&broadcast_stream->bap_stream, &broadcast_stream_ops);
+
+	stream_params.stream = &broadcast_source_stream;
+	stream_params.data_len = ARRAY_SIZE(bis_codec_data);
+	stream_params.data = bis_codec_data;
+
+	subgroup_param.stream_count = 1U;
+	subgroup_param.stream_params = &stream_params;
+	subgroup_param.codec_cfg = &broadcast_preset_48_2_1.codec_cfg;
+
+	create_param.subgroup_count = 1U;
+	create_param.subgroup_params = &subgroup_param;
+	create_param.qos = &broadcast_preset_48_2_1.qos;
+	create_param.packing = BT_ISO_PACKING_SEQUENTIAL;
+	create_param.encryption = false;
+
+	while (count < 2) {
```

Comment:
If both the sink and source relies on this value being the same, then I suggest to a `#define` and use that for both